stl/inc/shared_mutex

```diff
@@ -227,34 +227,34 @@ private:
 
 // CLASS TEMPLATE shared_lock
 template <class _Mutex>
-class _NODISCARD shared_lock { // shareable lock
+class shared_lock { // shareable lock
 public:
     using mutex_type = _Mutex;
 
     shared_lock() noexcept : _Pmtx(nullptr), _Owns(false) {}
 
-    explicit shared_lock(mutex_type& _Mtx)
+    _NODISCARD_CTOR explicit shared_lock(mutex_type& _Mtx)
         : _Pmtx(_STD addressof(_Mtx)), _Owns(true) { // construct with mutex and lock shared
         _Mtx.lock_shared();
     }
 
     shared_lock(mutex_type& _Mtx, defer_lock_t) noexcept
         : _Pmtx(_STD addressof(_Mtx)), _Owns(false) {} // construct with unlocked mutex
 
-    shared_lock(mutex_type& _Mtx, try_to_lock_t)
+    _NODISCARD_CTOR shared_lock(mutex_type& _Mtx, try_to_lock_t)
         : _Pmtx(_STD addressof(_Mtx)), _Owns(_Mtx.try_lock_shared()) {} // construct with mutex and try to lock shared
 
-    shared_lock(mutex_type& _Mtx, adopt_lock_t)
+    _NODISCARD_CTOR shared_lock(mutex_type& _Mtx, adopt_lock_t)
         : _Pmtx(_STD addressof(_Mtx)), _Owns(true) {} // construct with mutex and adopt ownership
 
     template <class _Rep, class _Period>
-    shared_lock(mutex_type& _Mtx, const chrono::duration<_Rep, _Period>& _Rel_time)
+    _NODISCARD_CTOR shared_lock(mutex_type& _Mtx, const chrono::duration<_Rep, _Period>& _Rel_time)
         : _Pmtx(_STD addressof(_Mtx)), _Owns(_Mtx.try_lock_shared_for(_Rel_time)) {
         // construct with mutex and try to lock for relative time
     }
 
     template <class _Clock, class _Duration>
-    shared_lock(mutex_type& _Mtx, const chrono::time_point<_Clock, _Duration>& _Abs_time)
+    _NODISCARD_CTOR shared_lock(mutex_type& _Mtx, const chrono::time_point<_Clock, _Duration>& _Abs_time)
         : _Pmtx(_STD addressof(_Mtx)), _Owns(_Mtx.try_lock_shared_until(_Abs_time)) {
         // construct with mutex and try to lock until absolute time
     }

```

Comment:
`shared_lock(shared_lock&& _Other)` below isn't being marked, but `unique_lock`'s move ctor was.

Reply:
I didn't mark it because it's not with the other constructors, which we should fix. I'll push into this and the internal PR.

Reply:
Ugh, even the synopsis order for `shared_lock` in the Standard is an abomination.